FW-5232 Add hasTranslation to search API

NolanVH opened the pull request:
### Description of Changes
This PR adds the boolean `hasTranslation` parameter to the search API.

A true value is added to the `has_translation` index as follows:
-  if a dictionary entry has any related translations
- if a song has non empty and non null title translation
- if a story has a non emptry and non null title translation

### Checklist
- ~README / documentation has been updated if needed~
- [x] PR title / commit messages contain Jira ticket number if applicable
- [x] Tests have been updated / created if applicable
- ~Admin Panel has been updated if models have been added or modified~
- ~Migrations have been updated and committed if applicable~
- [ ] Team Postman workspace has been updated if applicable

### Reviewers Should Do The Following:
- [x] Review the code changes here
- [ ] Pull the branch and test locally

### Additional Notes
We will need to re-index on release.

sarahfirstvoices commented:
I'm on the fence about the code smells. On the one hand, since none of the parameters are required and they are always passed by name, it could be reasonable to use **kwargs in the function definition. On the other hand, the method only supports this specific set of parameters, not any arbitrary ones, so it makes more sense to define them explicitly. For now I would lean toward just ignoring the code smells

NolanVH commented on firstvoices/backend/search/query_builder.py:
```diff
@@ -95,4 +97,7 @@ def get_search_query(
 
+    if has_translation is not None:
+        search_query = search_query.query(get_has_translation_query(has_translation))
+
```
Good call, thanks. I have updated the function.

NolanVH commented:
> I'm on the fence about the code smells. On the one hand, since none of the parameters are required and they are always passed by name, it could be reasonable to use **kwargs in the function definition. On the other hand, the method only supports this specific set of parameters, not any arbitrary ones, so it makes more sense to define them explicitly. For now I would lean toward just ignoring the code smells

Thanks for the feedback. I have removed the code smells and added a comment stating that we will refactor the function in [FW-5273](https://firstvoices.atlassian.net/browse/FW-5273).
